### PR TITLE
Added support for using default credentials instead of an access token for authentication.

### PR DESCRIPTION
**Pull request by icnocop**
Fixes issue #3

### CODE REVIEW

**Comment by icnocop** [comment]
Do you think adding a dependency on https://www.nuget.org/packages/semver/ will be an issue?

**Comment by daveaglick** [comment]
I wouldn't think so - so you can parse out the current TFS version?

BTW - thanks for all the leg work on this! Let me know when it's out of draft and I'll try to quickly review and turn it around.

**Comment by icnocop** [comment]
Right, so the `api-version` string can be parsed and determine which JSON schema to use when sending requests to the TFS API.

**Comment by icnocop** [comment]
@daveaglick, this is now ready for your review. :)

I've verified this version works on TFS 2015 Update 4, just like the "Publish Test Results" build task, using the following parameters:
`/logger:AzurePipelines;Verbose=true;UseDefaultCredentials=True;ApiVersion=3.0-preview.2;GroupTestResultsByClassName=False`

The only things I've noticed that are different so far:

1. A trx file is not attached to each test result.
2. The error messages have a different format.
    They include STDERR and STDOUT and I think otherwise those are only included in the trx file.

I noticed that long error messages are cut off with a maximum of about 500 characters, but I'm not sure if the behavior is the same when using the "Publish Test Results" build task for example.

Thank you.

**Comment by icnocop** [comment]
I went ahead and added support for uploading console outputs and errors as test result attachments and also support for uploading test result files.

**Comment by daveaglick** [comment]
Awesome! Hoping to get some much-needed OSS time to get this merged and deployed tomorrow )as long as nothing unexpected crops up).

**Comment by icnocop** [comment]
In regards to https://developercommunity.visualstudio.com/content/idea/409015/show-all-tests-in-the-hierarchy-in-test-summary.html, has it been fixed already?

The screenshots in the docs seem to indicate the test summary is calculated from all test results in all test runs even though there's a note that still reads "Metrics in the test summary section, such as the total number of tests, passed, failed, or other are computed using the root level of the summarized test result.".

See the first screenshot in https://docs.microsoft.com/en-us/azure/devops/pipelines/test/review-continuous-test-results-after-build?view=azure-devops#tests-tab.

I'm not sure if the issue has been fixed and the docs just need to be updated.

For example, if you disable grouping test results by their class name when publishing, and you group the test results by their `Container`, does it show the expected test summary details?

**Comment by icnocop** [comment]
Hi @daveaglick .

I noticed this PR was closed but the code was unmerged. 

Was there a reason?

Thank you.

**Comment by daveaglick** [comment]
Whops! Unintended consequence of changing the main branch name. I'll reopen and retarget this - I still intend to get to it. Thanks for the call out.

**Comment by daveaglick** [comment]
This is a crazy awesome PR, thanks a ton for all the legwork. I use this library in my projects, but I haven't given it much attention lately so I'm happy for the assist. Have any interest in being a co-maintainer?

File: ReleaseNotes.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# 1.0.4
+
+- [Feature] Added support for using default credentials instead of an access token for authentication.
```
Comment (daveaglick) [inline ReleaseNotes.md:3]: I'll add this after the merge and bump it to 1.1.0 to reflect the magnitude of the changes here

**Comment by daveaglick** [comment]
> For example, if you disable grouping test results by their class name with this logger, or use a classic logger and publisher, and you group the test results by their Container, does it show the expected test summary details?

Good question - I haven't been following along, but this nesting problem really annoyed me when I first started working on it. It's been a while so maybe there's hope.

**Comment by icnocop** [comment]
Thank you Dave.

Yes, I can try to help maintain this repository if/when needed.

Thank you. 😊

**Pull request merged**